source/dub/dependency.d

```diff
@@ -273,7 +295,12 @@ struct Dependency {
 	*/
 	string toString()()
 	const {
-		auto ret = versionSpec;
+		string ret;
+
+		if (!repository.empty) {
+			ret ~= "#";
+		}
+		ret ~= versionSpec;
```

Comment:
depending on where this toString value is presented to the user, it might be worth implementing parsing such a path pattern so for example `"foo": "https://github.com/bla.git#abc"` would work which would be epic.

However I think this might also fit in another PR.

Reply:
I personally don't like it that much. It is something npm.js does and I find separate fields more readable and more informative.

But if it has to be implemented, I'd prefer a separate PR. I'd like to make as few changes as possible to see if this feature has any future at all.